test/new-e2e/processes/linux_test.go

```diff
@@ -0,0 +1,227 @@
+	s.ddAPIClient = common.NewAPIClient(configuration)
+	s.procAPIClient = datadog.NewProcessesApi(s.ddAPIClient)
+}
+
+func (s *LinuxTestSuite) TearDownSuite() {
+	s.ec2.Close()
+}
+
+func (s *LinuxTestSuite) TearDownTest() {
+	killAndRemoveContainers(s.T(), s.ec2.sshClient)
+	s.T().Log("uninstalling the datadog agent")
+	stdout, err := clients.ExecuteCommand(s.ec2.sshClient, "sudo apt-get remove --purge datadog-agent -y")
+	if err != nil {
+		s.T().Logf("error uninstalling the datadog agent. stdout:%s, err: %v", stdout, err)
+	}
+}
+
+func (s *LinuxTestSuite) TestProcessAgentOnLinux() {
+	hostName := createHostName(s.T().Name())
+	hostTag := fmt.Sprintf("host:%s", hostName)
+
+	s.T().Logf("start the datadog agent with hostname: %s", hostName)
+	// install the agent
+	stdout, err := clients.ExecuteCommand(s.ec2.sshClient, fmt.Sprintf(
+		"REPO_URL=datad0g.com"+
+			" DD_SITE=\"datadoghq.com\""+
+			" DD_AGENT_MAJOR_VERSION=%d"+
+			" DD_API_KEY=%s"+
+			" DD_AGENT_DIST_CHANNEL=%s"+
+			" bash -c \"$(curl -L https://s3.amazonaws.com/dd-agent/scripts/install_script.sh)\"",
+		agentMajorVersion, s.ec2.ddAPIKey, agentDistChannel),
+	)
+	s.T().Log(stdout)
+	require.NoError(s.T(), err)
+
+	// Setup Process Config
+	ddCfg := agentConfig{
+		APIKey:      s.ec2.ddAPIKey,
+		Hostname:    hostName,
+		LogsEnabled: true,
+		ProcessCfg: ProcessConfig{
+			Enabled: true,
+		},
+	}
+
+	cfgYaml, err := yaml.Marshal(&ddCfg)
+	require.NoError(s.T(), err)
+	s.T().Logf("%s", cfgYaml)
+	_, err = clients.ExecuteCommand(s.ec2.sshClient, "sudo chmod o+rw /etc/datadog-agent/datadog.yaml")
+	require.NoError(s.T(), err)
+
+	file, err := s.sftpClient.OpenFile("/etc/datadog-agent/datadog.yaml", os.O_WRONLY|os.O_CREATE|os.O_TRUNC)
+	require.NoError(s.T(), err)
+	_, err = file.Write(cfgYaml)
+	require.NoError(s.T(), err)
+
+	// Setup System Probe Config
+	sysProbeCfg := probeConfig{
+		SysProbeCfg: SystemProbeConfig{
+			Enabled: true,
+			ProcCfg: ProcessConfig{
+				Enabled: true,
+			},
+		},
+		NetworkCfg: NetworkConfig{
+			Enabled: true,
+		},
+	}
+
+	sysProbeYaml, err := yaml.Marshal(&sysProbeCfg)
+	require.NoError(s.T(), err)
+	s.T().Logf("%s", sysProbeYaml)
+	_, err = clients.ExecuteCommand(s.ec2.sshClient, "sudo touch /etc/datadog-agent/system-probe.yaml")
+	require.NoError(s.T(), err)
+	_, err = clients.ExecuteCommand(s.ec2.sshClient, "sudo chmod o+rw /etc/datadog-agent/system-probe.yaml")
+	require.NoError(s.T(), err)
+
+	file, err = s.sftpClient.OpenFile("/etc/datadog-agent/system-probe.yaml", os.O_WRONLY|os.O_CREATE|os.O_TRUNC)
+	require.NoError(s.T(), err)
+	_, err = file.Write(sysProbeYaml)
+	require.NoError(s.T(), err)
+
+	// Restart Process Agent for new agentConfig
+	_, err = clients.ExecuteCommand(s.ec2.sshClient, "sudo service datadog-agent restart")
+	require.NoError(s.T(), err)
+
+	s.waitForProcessAgent()
+
+	// Start containers
+	stdout, err = clients.ExecuteCommand(s.ec2.sshClient,
+		"docker run --name spring-hello -d springio/gs-spring-boot-docker")
```

Comment:
We are pulling `latest` here from dockerhub, and actually looking at it, it's not tagged in dockerhub. To avoid hitting the docker limits we should probably use GCR/ECR and use a specific tag to prevent accidental changes to this image. It might be worth having these copied into DD-owned repo to address issues with unknown code.